src/RestSharp/RestClient.cs

```diff
@@ -108,6 +74,57 @@ public partial class RestClient : IDisposable {
     /// <param name="baseUrl"></param>
     public RestClient(string baseUrl) : this(new Uri(Ensure.NotEmptyString(baseUrl, nameof(baseUrl)))) { }
 
+    public RestClient(HttpClient httpClient, RestClientOptions? options = null, bool disposeHttpClient = false) {
+        if (Options?.CookieContainer != null) {
+            throw new ArgumentException("Custom cookie container cannot be added to the HttpClient instance", nameof(options.CookieContainer));
+        }
+
+        UseDefaultSerializers();
+
+        HttpClient         = httpClient;
+        Options            = options ?? new RestClientOptions();
+        CookieContainer    = new CookieContainer();
+        _disposeHttpClient = disposeHttpClient;
+
+        ConfigureHttpClient(HttpClient);
+    }
+
+    /// <summary>
+    /// Creates a new instance of RestSharp using the message handler provided. By default, HttpClient disposes the provided handler
+    /// when the client itself is disposed. If you want to keep the handler not disposed, set disposeHandler argument to false.
+    /// </summary>
+    /// <param name="handler">Message handler instance to use for HttpClient</param>
+    /// <param name="disposeHandler">Dispose the handler when disposing RestClient, true by default</param>
+    public RestClient(HttpMessageHandler handler, bool disposeHandler = true) : this(new HttpClient(handler, disposeHandler)) { }
```

Comment:
I think this should call the other constructor with `, true` to ensure that the constructed `HttpClient` no matter if the injected `handler` should be disposed.

Reply:
Yeah, you're right. Changed to
```csharp
public RestClient(HttpMessageHandler handler, bool disposeHandler = true) : this(new HttpClient(handler, disposeHandler), null, true) { }
```